app/Http/Requests/BulkUserCreateRequest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class BulkUserCreateRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $circle_id = $this->route('circle_id');
+
+        return [
+            'users' => ['required', 'array', 'min:1'],
+            'users.*.address' => ['required', 'string', 'size:42', 'distinct',
+                // validate id is unique within circle
+                Rule::unique('users')->where(function ($query) use ($circle_id) {
+                    return $query->where('circle_id', $circle_id)->whereNull('deleted_at');
+                })],
+            'users.*.name' => ['required', 'string', 'max:255'],
```

Comment:
Not sure if easy to do, but ideally I'd think all these validation rules are defined in exactly one spot not across multiple places (https://github.com/coordinape/coordinape-backend/blob/main/app/Http/Requests/UserRequest.php#L40) but may not be easy to do especially as this is wrapped within the "users" context here vs not in the UserRequest file.

Reply:
idk maybe i can make a constant for every rule. but might result in a lot of constants created

Reply:
nah, this is fine then.